### PR TITLE
Support for Scala-2.13

### PR DESCRIPTION
**Pull request by FabioPinheiro**
This PR is branched from https://github.com/mfirry/shocon/tree/fastparse2
Based on the work from @mfirry and @andreaTP  to updated  the FastParse liberty to version 2.X

Note: For what I know there will be a follow up PR to support scala 2.13
(In the meantime I'm using the compilation from this PR to advance on other fronts)

### CODE REVIEW

File: shared/src/main/scala/org/akkajs/shocon/SHocon.scala [hunk 1]
```diff
@@ -22,7 +23,7 @@ package object shocon extends Extractors {
       lazy val unwrapped = elements.map(_.unwrapped)
     }
     case class Object(fields: Map[Key, Value]) extends Value {
-      lazy val unwrapped = fields.mapValues(_.unwrapped)
+      lazy val unwrapped = fields.view.mapValues(_.unwrapped).toSeq
```
Comment (FabioPinheiro) [inline shared/src/main/scala/org/akkajs/shocon/SHocon.scala]: I added the toSeq make it compile on:
`shocon/facade/shared/src/main/scala/com/typesafe/config/Config.scala:54:52: value toMap is not a member of Any`
This was not compiling for scala 2.12 only. I think `scala-collection-compat` is not covering this case.
I don't no if it's by design and I should add create some implicit or a defect not yet covered.
@andreaferretti can you have a look? Maybe it's worth opening the bug there.

File: shared/src/main/scala/org/akkajs/shocon/SHocon.scala [hunk 2]
```diff
@@ -109,8 +110,11 @@ package object shocon extends Extractors {
                 case _ =>
                   k -> v
               }
-          } ++ mergeable.fields.filterKeys(diff.contains)
-
+          } ++ mergeable.fields.view.filter(e => diff.contains(e._1))
```
Comment (FabioPinheiro) [inline shared/src/main/scala/org/akkajs/shocon/SHocon.scala]: This was also not compiling for scala 2.12 only.
I think `scala-collection-compat` is not covering this case.

File: shared/src/main/scala/org/akkajs/shocon/SHocon.scala (hunk 1 above)
Comment (andreaTP) [reply shared/src/main/scala/org/akkajs/shocon/SHocon.scala]: the idiomatic way of doing this should be:
```
fields.view.map(_.unwrapped).to(Seq)
```
according to:
https://www.scala-lang.org/blog/2017/02/28/collections-rework.html

**Comment by andreaTP** [comment]
thanks a lot @FabioPinheiro !
I'll take a closer look and if the CI is happy I merge!

File: shared/src/main/scala/org/akkajs/shocon/SHocon.scala (hunk 1 above)
Comment (FabioPinheiro) [reply shared/src/main/scala/org/akkajs/shocon/SHocon.scala]: ah. has been trying to figure out what the difference was between toSet and to(Seq)

**Comment by FabioPinheiro** [comment]
We should squash this into one commit
But updated the FastParse liberty to version 2.X should be another commit or not? (fwi @mfirry)

**Comment by andreaTP** [comment]
not strict squashing in this repo, no big worries.

File: project/Common.scala [hunk 3]
```diff
@@ -6,10 +6,10 @@ object Common extends AutoPlugin {
   override lazy val projectSettings = super.projectSettings ++ Seq(
     name := "shocon",
     organization := "org.akka-js",
-    version := "0.4.1",
-    scalaVersion := "2.11.11",
+    version := "0.5.0-SNAPSHOT",
+    scalaVersion := "2.13.0",
     crossScalaVersions :=
-      Vector("2.11.11", "2.12.4"),
+      Vector(/*"2.11.11",*/ "2.12.8", "2.13.0"),
```
Comment (FabioPinheiro) [inline project/Common.scala]: Scala 2.12.9 is out
Should we clean up the commented configs about Scala Native (and 2.11.11)

File: .travis.yml [hunk 4]
```diff
@@ -6,17 +6,17 @@ jdk: oraclejdk8
 language: scala
 
 scala:
-  - "2.12.4"
-  - "2.11.11"
```
Comment (andreaTP) [inline .travis.yml]: can we keep 2.11 ?

File: project/Common.scala [hunk 5]
```diff
@@ -6,10 +6,10 @@ object Common extends AutoPlugin {
   override lazy val projectSettings = super.projectSettings ++ Seq(
     name := "shocon",
     organization := "org.akka-js",
-    version := "0.4.1",
-    scalaVersion := "2.11.11",
+    version := "0.5.0-SNAPSHOT",
+    scalaVersion := "2.13.0",
     crossScalaVersions :=
-      Vector("2.11.11", "2.12.4"),
+      Vector("2.12.8", "2.13.0"),
```
Comment (andreaTP) [inline project/Common.scala]: I think we can still keep 2.11, isn't it?

**Comment by andreaTP** [comment]
Ok, fastparse itself dropped 2.11 support, that's already a good enough motivation to do the same:
https://mvnrepository.com/artifact/com.lihaoyi/fastparse

**Comment by andreaTP** [comment]
thanks a lot @FabioPinheiro and @mfirry for the help here!

**Pull request merged**